code/game/objects/items/weapons/tools.dm

```diff
@@ -396,25 +405,57 @@
 /obj/item/weapon/weldingtool/hugetank
 	name = "upgraded welding tool"
 	max_fuel = 80
-	w_class = 3.0
+	w_class = 2.0
 	matter = list(DEFAULT_WALL_MATERIAL = 70, "glass" = 120)
 	origin_tech = "engineering=3"
 
+
+
+
+//The Experimental Welding Tool!
 /obj/item/weapon/weldingtool/experimental
 	name = "experimental welding tool"
+	desc = "A scientifically-enhanced welding tool that uses fuel-producing microbes to gradually replenish its fuel supply"
 	max_fuel = 40
-	w_class = 3.0
+	w_class = 2.0
 	matter = list(DEFAULT_WALL_MATERIAL = 70, "glass" = 120)
-	origin_tech = "engineering=4;phorontech=3"
+	origin_tech = "engineering=4;biotech=4"
 	var/last_gen = 0
+	var/fuelgen_delay = 800//The time, in deciseconds, required to regenerate one unit of fuel
+	//800 = 1 unit per 1 minute and 20 seconds,
+	//This is roughly half the rate that fuel is lost if the welder is left idle, so it you carelessly leave it on it will still run out
+
+//Make sure the experimental tool only stops processing when its turned off AND full
+/obj/item/weapon/weldingtool/experimental/set_processing(var/state = 0)
+	if (state == 1)
+		processing_objects.Add(src)
+		last_gen = world.time
+	else if (welding == 0 && get_fuel() >= max_fuel)

```

Comment:
This is dangerous.

There's a possibility of the tool's welding being set to `0`, but it remaining in the processing queue because the second condition is not filled.

Whenever a welding tool is `qdel()`-d, it does the following:
```
if (welding)
    processing_objects.Remove(src)
```

This is to prevent runtime errors from the Obj processor as it tries to call up `null.process()`. You should probably alter the experimental welding tool to have a `var/processing`, and its child procs to run off of that, instead of `var/welding`.